### PR TITLE
Add issue templates

### PR DESCRIPTION
**Pull request by aholstrup1**
Adding issue templates ahead of moving BCApps open-source.

When opening an issue you get this page:
image

The only types of issues you can create are Bug reports, issues related to BCIdeas and vulnerabilities. The remaining options like "Feature requests" takes you to BCideas. Reporting a product defect with customer impact takes you to aka.ms/bcsupport
image

image

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bcidea.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Create issue for BC Idea I want to implement
+description: Create issue for BC Idea you want to implement
+title: "[BC Idea]: "
+labels: ["bcidea"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to create a BC Idea issue!
```
Comment (mazhelez) [inline .github/ISSUE_TEMPLATE/bcidea.yml]: https://aka.ms/bcideas is needed around here.

File: .github/ISSUE_TEMPLATE/bcidea.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      placeholder: Description of BC Idea
+    validations:
+      required: true
+  - type: dropdown
+    id: category
+    attributes:
+      label: BC Idea Category
+      description: What category is the BC Idea created under?
+      options:
+        - No Category
```
Comment (mazhelez) [inline .github/ISSUE_TEMPLATE/bcidea.yml]: Most of these categories are not for apps in this repo.

@JesperSchulz are these categories really needed here? Can we adapt them for _System Application_?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+title: "[Bug]: "
+labels: ["bug"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this bug report! 
+
+        ⚠️⚠️ BEFORE WE GET STARTED ⚠️⚠️
+        This is not the right place to report product defects with customer impact to Microsoft! Issues reported in this repository do not fall under SLAs (Service Level Agreements) and hence have no guaranteed time to mitigation. If your customers are experiencing product defects or you have discovered a severe issue in the product, please follow the steps outlined in https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/technical-support to get your issue routed to the right team at Microsoft and to get it treated with the right priority.
```
Comment (mazhelez) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        This is not the right place to report product defects with customer impact to Microsoft! Issues reported in this repository do not fall under SLAs (Service Level Agreements) and hence have no guaranteed time to mitigation. If your customers are experiencing product defects or you have discovered a severe issue in the product, please follow the steps outlined in https://learn.microsoft.com/dynamics365/business-central/dev-itpro/technical-support to get your issue routed to the right team at Microsoft and to get it treated with the right priority.
```

File: .github/ISSUE_TEMPLATE/bcidea.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+name: Create issue for BC Idea I want to implement
```
Comment (mazhelez) [inline .github/ISSUE_TEMPLATE/bcidea.yml]: Can the files have the .yaml extension instead? Please :)

File: .github/ISSUE_TEMPLATE/bcidea.yml (hunk 2 above)
Comment (aholstrup1) [reply .github/ISSUE_TEMPLATE/bcidea.yml]: I'm also completely fine with removing them btw. Maybe I went a bit overboard with these issue templates 😆

File: .github/ISSUE_TEMPLATE/bcidea.yml (hunk 4 above)
Comment (aholstrup1) [reply .github/ISSUE_TEMPLATE/bcidea.yml]: I've updated bcidea and bug_report to use yaml. Unfortunately, config.yml doesn't work if the extension is yaml :( 
https://github.com/orgs/community/discussions/36812

File: .github/ISSUE_TEMPLATE/bcidea.yml (hunk 2 above)
Comment (mazhelez) [reply .github/ISSUE_TEMPLATE/bcidea.yml]: Let's remove them for now.

File: .github/ISSUE_TEMPLATE/bcidea.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        Thanks for taking the time to create a BC Idea issue!
+
+        ⚠️⚠️ BEFORE WE GET STARTED ⚠️⚠️
+        Please do not create an issue for a BC Idea unless you intend to implement it yourself. If you are not planning to implement the BC Idea yourself, please go to aka.ms/bcideas and upvote the feature on BC Ideas instead. 
+
+        Before you create a new issue, please check the following:
+        🔎 Search existing issues to avoid creating duplicates.
+        🧪 Test using the latest bcinsider builds to see if your issue has already been fixed
+
+        Read more about what and how to contribute in the CONTRIBUTIONS document of this repository: https://github.com/microsoft/BCApps/blob/main/CONTRIBUTING.md.
```
Comment (JesperSchulz) [inline .github/ISSUE_TEMPLATE/bcidea.yaml:17]: The CONTRIBUTING.md file doesn't exist in this repo, does it? We should add it - and update it to reflect these process adjustments.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+name: Bug Report
```
Comment (JesperSchulz) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: Let's call this: "Report and address a bug"

File: .github/ISSUE_TEMPLATE/bcidea.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+name: Create issue for BC Idea I want to implement
```
Comment (JesperSchulz) [inline .github/ISSUE_TEMPLATE/bcidea.yaml]: Let's call this: "Suggest a feature for contribution"

File: .github/ISSUE_TEMPLATE/bcidea.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+description: Create issue for BC Idea you want to implement
```
Comment (JesperSchulz) [inline .github/ISSUE_TEMPLATE/bcidea.yaml]: Suggest a feature you would like to implement, which is based on a BC idea.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+description: Create a bug report to help us improve
+title: "[Bug]: "
```
Comment (JesperSchulz) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:3]: Create a bug report for a product defect, you would like to address yourself.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Feature requests
```
Comment (JesperSchulz) [inline .github/ISSUE_TEMPLATE/config.yml]: Request a feature

File: .github/ISSUE_TEMPLATE/config.yml [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+    url: http://aka.ms/bcideas
+    about: Please go to http://aka.ms/bcideas to log your idea. Creating an issue here is not the right way.
+  - name: Report a product defect that is urgent and impacts customers
```
Comment (JesperSchulz) [inline .github/ISSUE_TEMPLATE/config.yml]: Report a bug which has customer impact

File: .github/ISSUE_TEMPLATE/bcidea.yaml (hunk 5 above)
Comment (aholstrup1) [reply .github/ISSUE_TEMPLATE/bcidea.yaml:17]: Created

File: .github/ISSUE_TEMPLATE/bcidea.yaml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to create a BC Idea issue!
+
+        ⚠️⚠️ BEFORE WE GET STARTED ⚠️⚠️
+        Please do not create an issue for a BC Idea unless you intend to implement it yourself. If you are not planning to implement the BC Idea yourself, please go to aka.ms/bcideas and upvote the feature on BC Ideas instead. 
+
+        Before you create a new issue, please check the following:
```
Comment (mazhelez) [inline .github/ISSUE_TEMPLATE/bcidea.yaml]: Is this part needed here? "Test using the latest bcinsider builds to see if your issue has already been fixed" doesn't make much sense for a feature contribution...

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
+# Contributing
+
+## What to contribute
+
+This project primarily welcomes contributions of two types:
+
+- Pull Requests for pre-approved issues, which are based on BC Ideas (http://aka.ms/bcideas) that have been picked as candidates for code contribution projects.
```
Comment (mazhelez) [inline CONTRIBUTING.md]:
```suggestion
- Pull Requests for pre-approved features, which are based on BC Ideas (http://aka.ms/bcideas) that have been picked as candidates for code contribution projects.
```

Also, how does one know which BC Ideas have been "picked as candidates for code contribution projects"?

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+- Pull Requests based on approved issues, which address smaller non-customer reported improvements and fixes.
```
Comment (mazhelez) [inline CONTRIBUTING.md]: Wording could be better here.

```suggestion
- Pull Requests with fixes and improvements addressing pre-approved issues, which are non-customer reported.
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+In either case, an approved issue in state "approved", must exist, before a Pull Request can be created. **Pull Requests without link to approved issues will instantly get rejected.**
```
Comment (mazhelez) [inline CONTRIBUTING.md]: Is it state "approved" or a tag?
C'mon, will we really reject PRs if there's no issue linked?

A comment with "link an issue, please" + a tag "missing linked issue" is enough, isn't it?

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+    Now a Pull Request can be created. It is important that the Pull Request is linked to an approved issue. **Pull Requests which aren't linked to approved issues will get rejected.** 
+    
+   If it is your first contribution, you are required to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution. For details, visit https://cla.microsoft.com. When you submit a pull request, a CLA-bot will automatically determine whether you need
+    to provide a CLA and decorate the Pull Request appropriately (e.g., label, comment). Simply follow the instructions provided by the bot. You will only need to do this once across all repositories using our CLA.
+
+
+4. **Get a succesful build and code review approval**
+
+    Before your Pull Request is merged, it must have a succesful Pull Request Build and pass a code review. This code review is done by both Microsoft as well as members of the community. This ensures that quality of the code is on par with Microsoft's and the community's standards.
```
Comment (mazhelez) [inline CONTRIBUTING.md]:
```suggestion
    Before your Pull Request can be merged, it must have successful required status checks, all conversations resolved and an approval from a code owner. This code review can be done by both Microsoft and members of the community. This ensures that quality of the code is on par with Microsoft's and the community's standards.
```

File: CONTRIBUTING.md (hunk 15 above)
Comment (aholstrup1) [reply CONTRIBUTING.md]: Changed the wording to
In either case, **an approved GitHub issue in state "approved", must exist, before a Pull Request can be created**. Once a Pull Request is created it **must** be linked to a GitHub issue.

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+- Pull Requests with fixes and improvements addressing pre-approved issues, which are non-customer reported.
+
+In either case, **an approved GitHub issue in state "approved", must exist, before a Pull Request can be created**. Once a Pull Request is created it **must** be linked to a GitHub issue. 
+
+**⚠IMPORTANT⚠:** This is not the right place to report product defects with customer impact to Microsoft! Issues created in this repository won't get picked up by the Microsoft engineering team and issues reported in this repository do not fall under SLAs (Service Level Agreements) and hence have no guaranteed time to mitigation, just as provided fixes won't get backported to all supported versions of the product.
+
+If your customers are experiencing product defects or you have discovered a severe issue in the product, please follow the steps outlined in ["Technical Support for Dynamics 365 Business Central"](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/technical-support) to get your issue routed to the right team at Microsoft and to get it treated with the right priority.
```
Comment (mazhelez) [inline CONTRIBUTING.md]:
```suggestion
If your customers are experiencing product defects or you have discovered a severe issue in the product, please follow the steps outlined in ["Technical Support for Dynamics 365 Business Central"](https://docs.microsoft.com/dynamics365/business-central/dev-itpro/technical-support) to get your issue routed to the right team at Microsoft and to get it treated with the right priority.
```

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+    Once the issue is created on GitHub, it still needs to be approved. Once approved, the rest of the process is identical to the process described in **A)**.
+
+## More Resources
+* This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/)
+or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+* If you are a beginner at contributing, start by following the [GitHub guidelines](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+
+* If you are new to developing for Business Central, visit [Developers Learning Catalog](https://docs.microsoft.com/en-us/dynamics365/business-central/readiness/readiness-learning-developers).
```
Comment (mazhelez) [inline CONTRIBUTING.md]:
```suggestion
* If you are new to developing for Business Central, visit [Developers Learning Catalog](https://docs.microsoft.com/dynamics365/business-central/readiness/readiness-learning-developers).
```

**Pull request merged**